### PR TITLE
feat(auth): purge session on mid-session 401 (token expiry)

### LINKED ISSUE #1130: P0 — Real authentication flow (live login/register/me, Bearer token, 401 handling)
Wire the always-live auth calls to Fly.io and make the token usable by every JWT screen.

**Endpoints:** `POST /auth/login`, `POST /auth/register`, `POST /auth/forgot-password`, `GET /auth/me`.
**Scope:** persist token, attach `Authorization: Bearer`, handle 401/expiry, coexist with the demo trigger-credentials path.

**Done when:** a real login on device returns a token, persists it, and a JWT GET returns 200 in the `preview` build.

Part of #1128.

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Wires global handling of a mid-session **401** (expired/invalidated JWT) so the app
no longer leaves the user "logged in" with a dead token. Any authenticated request that
returns 401 now clears the persisted session and bounces the user back to sign-in.

This is the last missing piece of #1130: the auth chain (`login` / `register` /
`getCurrentUser` / token storage / `Authorization: Bearer`) was already implemented and
verified live against Fly.io — only token-expiry handling was absent.

## Context

- `session.ts` — registerable unauthorized-handler (`setUnauthorizedHandler` /
  `notifyUnauthorized`), kept on the low-level module so the HTTP client never imports React.
- `http-client.ts` — invoke `notifyUnauthorized()` on a 401 **only for authenticated
  requests** (`auth === true`). A 401 from login with bad credentials (`auth: false`)
  stays a normal failure — no logout/redirect.
- `auth-context.tsx` — register the handler to clear the session and surface a FR message;
  no-op in demo mode; cleaned up on unmount.

Verified end-to-end against `https://brasse-bouillon-api.fly.dev` (register → login →
`/auth/me` with Bearer → `DELETE /auth/me`): full cycle returns 201/200/200/200 and the
throwaway test account was deleted (re-login → 401, prod left clean).

Part of #1128.

## Checklist

- [x] `npm run ci:check` green (lint + typecheck + format)
- [x] Auth test surface green — 58/58 across 5 suites (H/S/E)
- [x] New behaviour covered: authenticated 401 purges session; `auth:false` 401 does not; demo mode no-op
- [x] Live contract verified against Fly.io
- [x] No `any`, named exports only, no direct `fetch` outside the http client

Closes #1130.

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@Smith06S @Thais9723 @Fabien-Ori — Pour info : l'application déconnecte désormais proprement un utilisateur dont la session a expiré.

Quand on reste connecté longtemps, le « laissez-passer » de connexion (le jeton qui prouve qu'on est identifié) finit par expirer côté serveur. Jusqu'ici, l'appli ne s'en rendait pas compte : l'utilisateur semblait connecté mais ses actions échouaient en silence. Désormais, dès qu'une action est refusée pour cause de session expirée, l'appli efface la session et renvoie automatiquement vers l'écran de connexion (avec un message « Session expirée, reconnecte-toi »). Important : une simple erreur de mot de passe à la connexion ne déclenche pas ça — seul un vrai jeton périmé le fait.

Rien à faire de votre côté. C'est une brique de la connexion réelle au serveur (épopée #1128) : l'authentification est maintenant testée et validée en conditions réelles.

File: packages/mobile-app/src/core/http/http-client.ts [hunk 1]
```diff
@@ -67,6 +67,15 @@ export async function request<T>(
   const payload = await parseBody(response);
 
   if (!response.ok) {
+    // A 401 on an *authenticated* request means the token expired or was
+    // invalidated mid-session — notify so the session can be purged. Gated
+    // on `auth` so a 401 from an unauthenticated call (e.g. login with bad
+    // credentials, which passes `auth: false`) is treated as a normal
+    // failure and never triggers a logout/redirect.
+    if (response.status === 401 && auth) {
+      authSession.notifyUnauthorized();
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/core/http/http-client.ts]: Good catch — fixed in 0aac70a. The 401 path now only calls `notifyUnauthorized()` when the token attached to the request is still the current one (`attachedToken && attachedToken === authSession.getAccessToken()`). A stale in-flight request carrying a token from before a logout/re-login no longer purges the fresh session. Added an edge test covering the logout/re-login race.

**Comment by benoit-bremaud** [comment]
## Review summary — ready to merge

**Scope:** real auth against the live Fly.io API (#1130) + mid-session 401/token-expiry handling, hardened for the logout/re-login race.

**Commits**
- `6a46629` — purge session on a mid-session 401 (global unauthorized handler: `session.ts` → `http-client.ts` → `auth-context.tsx`), no-op in demo mode.
- `0aac70a` — only purge when the token that produced the 401 is still the current one (stale in-flight 401 no longer logs out a freshly re-authenticated user). Addresses the Codex review.

**CI (on `0aac70a`):** all green — mobile-app ✅, SonarCloud Code Analysis ✅, SonarCloud Scan ✅, security-audit ✅, changes ✅.

**Tests:** auth surface green, including the new edge case (stale-token 401 → no purge). H/S/E covered across http-client, auth-context and the session registry.

**Live verification:** register → login → `/auth/me` (Bearer) → `DELETE /auth/me` against `brasse-bouillon-api.fly.dev` returned 201/200/200/200; throwaway test account deleted (re-login → 401), no residue in prod.

**Reviews:** Codex P2 (stale-401 race) — fixed in `0aac70a`, replied inline, thread resolved. Copilot reviewer could not be attached (returns 422 — not a repo collaborator), so no Copilot review on this PR.

**Ready to merge.** Closes #1130; part of #1128.

**Comment by benoit-bremaud** [comment]
### Update — Copilot review complete

Copilot was re-triggered and reviewed all 6 changed files, **generating no comments** (it explicitly acknowledges the stale-token guard). This supersedes the earlier note about Copilot not being attached.

**Final gate status:**
- CI on `0aac70a` — all green ✅
- Codex P2 (stale-401 race) — fixed, replied, thread resolved ✅
- Copilot — reviewed, no comments ✅
- No open conversations.

**Ready to merge.** Closes #1130; part of #1128.
